x-pack/test/api_integration/apis/ml/filters/get_filters_stats.ts

```diff
@@ -0,0 +1,231 @@
+    model_plot_config: { enabled: true },
+  };
+
+  const testJobConfigs = [jobConfig1, jobConfig2];
+
+  const testDataList = [
+    {
+      filterId: 'ignore_a_airlines',
+      requestBody: {
+        description: 'Airlines starting with A',
+        items: ['AAL'],
+      },
+      expected: {
+        item_count: 1,
+        used_by: {
+          jobs: [jobConfig1.job_id, jobConfig2.job_id],
+          detectors: [
+            `${jobConfig1.analysis_config.detectors[0].detector_description} (${jobConfig1.job_id})`,
+            `${jobConfig2.analysis_config.detectors[1].detector_description} (${jobConfig2.job_id})`,
+          ],
+        },
+      },
+    },
+    {
+      filterId: 'ignore_b_airlines',
+      requestBody: {
+        description: 'Airlines starting with B',
+        items: ['BAA', 'BAB'],
+      },
+      expected: {
+        item_count: 2,
+        used_by: {
+          jobs: [jobConfig1.job_id],
+          detectors: [
+            `${jobConfig1.analysis_config.detectors[0].detector_description} (${jobConfig1.job_id})`,
+          ],
+        },
+      },
+    },
+    {
+      filterId: 'ignore_c_airlines',
+      requestBody: {
+        description: 'Airlines starting with C',
+        items: ['CAA', 'CAB', 'CCC'],
+      },
+      expected: {
+        item_count: 3,
+      },
+    },
+  ];
+
+  describe('get_filters_stats', function () {
+    before(async () => {
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      for (const testData of testDataList) {
+        const { filterId, requestBody } = testData;
+        await ml.api.createFilter(filterId, requestBody);
+      }
+
+      for (const job of testJobConfigs) {
+        await ml.api.createAnomalyDetectionJob(job);
+      }
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it(`should fetch all filters stats`, async () => {
+      const { body, status } = await supertest
+        .get(`/api/ml/filters/_stats`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS);
+      ml.api.assertResponseStatusCode(200, status, body);
+
+      expect(body).to.have.length(testDataList.length);
+
+      // Validate the contents of the stats response
+      for (const testData of testDataList) {
+        const { filterId, expected } = testData;
+
+        const actualFilterStats = body.find(
+          (filterStats: FilterStats) => filterStats.filter_id === filterId
+        );
+
+        expect(actualFilterStats).to.have.property('item_count').eql(expected.item_count);
+        if (expected.used_by !== undefined) {
+          expect(actualFilterStats).to.have.property('used_by');
+          expect(actualFilterStats.used_by).to.have.property('jobs');
+          expect(actualFilterStats.used_by.jobs).to.have.length(expected.used_by.jobs.length);
+          expect(actualFilterStats.used_by.jobs).to.eql(expected.used_by.jobs);
```

Comment:
With the recent changes, we could simplify this assertion:
```suggestion
          expect(actualFilterStats.used_by).to.have.property('jobs').eql(expected.used_by.jobs);
```